s3_management/manage.py

```diff
@@ -348,10 +350,14 @@ def from_S3(cls: Type[S3IndexType], prefix: str) -> S3IndexType:
                 for pattern in ACCEPTED_SUBDIR_PATTERNS
             ]) and obj.key.endswith(ACCEPTED_FILE_EXTENSIONS)
             if is_acceptable:
+                # Add PEP 503-compatible hashes to URLs to allow clients to avoid spurious downloads, if possible.
+                response = obj.meta.client.head_object(Bucket=BUCKET.name, Key=obj.key, ChecksumMode="ENABLED")
+                sha256 = response.get("ChecksumSHA256")
```

Comment:
Hi there @matteius, I believe this needs a little bit more polishing. The s3 checksum seems to be a base64 encoded byte value of the hash (judging by what's to be seen [here](https://github.com/pytorch/builder/pull/1433#discussion_r1261876922) or [here](https://github.com/aws/aws-cli/issues/6750#issuecomment-1619191575)), however what is required by the PEP 503 is a hex encoded value.
```suggestion
                response = obj.meta.client.head_object(Bucket=BUCKET.name, Key=obj.key, ChecksumMode="ENABLED")
                sha256 = (_b64 := response.get("ChecksumSHA256")) and base64.b64decode(_b64).hex()
```
This will also need an `import base64` at the top.

Reply:
Very good eye! :+1: (The first clue should've been the length :joy:)